falcon/util/uri.py

```diff
@@ -311,7 +314,7 @@ def parse_query_string(query_string, keep_blank_qs_values=False):
                     # will be the same for 'foo=1,,3' as 'foo=1&foo=&foo=3'.
                     v = [element for element in v if element]
```

Comment:
Technically, there's a breaking change here, although there isn't currently a test written for it. If individual elements are percent-encoded, they will no longer be decoded here. Do you think it would be worth doing the following? If so, it would be good to also add a test. I suspect few (if any) apps require this, but it may be better to be safe than sorry.

```python
if not keep_blank_qs_values:
    v = [decode(element) for element in v if element]
else:
    v = [decode(element) for element in v]
```

The alternative would be to add a "breaking change" notice to the commit message (and eventually the changelog when we release 0.3).